src/sql/parts/modelComponents/radioButton.component.ts

```diff
@@ -0,0 +1,119 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./radioButton';
+import {
+	Component, Input, Inject, ChangeDetectorRef, forwardRef, ComponentFactoryResolver,
+	ViewChild, ViewChildren, ElementRef, Injector, OnDestroy, QueryList, AfterViewInit
+} from '@angular/core';
+
+import * as sqlops from 'sqlops';
+import Event, { Emitter } from 'vs/base/common/event';
+
+import { ComponentBase } from 'sql/parts/modelComponents/componentBase';
+import { IComponent, IComponentDescriptor, IModelStore, ComponentEventType } from 'sql/parts/modelComponents/interfaces';
+import { RadioButton } from 'sql/base/browser/ui/radioButton/radioButton';
+import { CommonServiceInterface } from 'sql/services/common/commonServiceInterface.service';
+import { attachInputBoxStyler, attachListStyler } from 'vs/platform/theme/common/styler';
+
+@Component({
+	selector: 'radioButton',
+	template: `
+		<div class="modelview-radiobutton-container">
+		<span #input></span>
+			{{getLabel()}}
+		</div>
+	`
+})
+export default class RadioButtonComponent extends ComponentBase implements IComponent, OnDestroy, AfterViewInit {
+	@Input() descriptor: IComponentDescriptor;
+	@Input() modelStore: IModelStore;
+	private _input: RadioButton;
```

Comment:
In order for this component to be useful it should be a group of radio buttons that lets you choose between them, instead of just a single radio button. With the component representing a single radio button like this, extension authors will have to use multiple of them and manually hook up events to deselect the other buttons when one is selected. Instead, extension authors should be able to create a single component that contains multiple radio buttons that the user can choose from

Reply:
If we think that will be useful we should just accept `radio` as an input type for the input box component and make it work there instead of requiring an entirely new component

Reply:
as I said in another comment the properties for radio button are different than a text box

---
In reply to: [189119848](https://github.com/Microsoft/sqlopsstudio/pull/1439#discussion_r189119848) [](ancestors = 189119848)